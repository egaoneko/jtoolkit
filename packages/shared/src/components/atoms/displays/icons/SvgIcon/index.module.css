.svg {
  &.doubleExtraSmall {
    @apply text-xs;
  }

  &.extraSmall {
    @apply text-sm;
  }

  &.small {
    @apply text-lg;
  }

  &.medium {
    @apply text-2xl;
  }

  &.large {
    @apply text-4xl;
  }

  &.extraLarge {
    @apply text-6xl;
  }

  &.doubleExtraLarge {
    @apply text-8xl;
  }

  &.stroked {
    @apply stroke-current;

    *[stroke] {
      @apply stroke-current;
    }
  }

  &.filled {
    @apply fill-current;

    *[fill] {
      @apply fill-current;
    }
  }
}
