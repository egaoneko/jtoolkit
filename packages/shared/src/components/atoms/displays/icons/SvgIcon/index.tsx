import { forwardRef, SVGAttributes } from 'react';
import clsx from 'clsx';

import { Size } from 'src/enums/size';

import styles from './index.module.css';

export interface SvgIconProps extends SVGAttributes<SVGSVGElement> {
  size?: Size;
  stroked?: boolean;
  filled?: boolean;
  viewBox?: string;
  className?: string;
}

const SvgIcon = forwardRef<SVGSVGElement, SvgIconProps>(function SvgIcon(
  { size = Size.MEDIUM, stroked = false, filled = true, viewBox = '0 0 24 24', className, children, ...rest },
  ref,
) {
  return (
    <svg
      ref={ref}
      className={clsx(styles.svg, styles[size], { [styles.stroked]: stroked, [styles.filled]: filled }, className)}
      width="1em"
      height="1em"
      viewBox={viewBox}
      {...rest}
    >
      {children}
    </svg>
  );
});

export default SvgIcon;
