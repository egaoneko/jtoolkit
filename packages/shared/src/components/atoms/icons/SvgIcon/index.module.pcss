.svg {
  &.small {
    @apply text-sm;
  }

  &.medium {
    @apply text-base;
  }

  &.large {
    @apply text-lg;
  }

  &.stroked {
    @apply stroke-current;

    *[stroke] {
      @apply stroke-current;
    }
  }

  &.filled {
    @apply fill-current;

    *[fill] {
      @apply fill-current;
    }
  }
}
