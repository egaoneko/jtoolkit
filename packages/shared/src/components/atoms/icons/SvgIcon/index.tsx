import { FC, SVGAttributes } from 'react';
import clsx from 'clsx';

import { Size } from '../../../../enums/size';

import styles from './index.module.pcss';

export interface SvgIconProps extends SVGAttributes<SVGElement> {
  size?: Size;
  stroked?: boolean;
  filled?: boolean;
  viewBox?: string;
  className?: string;
}

const SvgIcon: FC<SvgIconProps> = ({
  size = Size.MEDIUM,
  stroked = false,
  filled = true,
  viewBox = '0 0 24 24',
  className,
  children,
  ...rest
}) => {
  return (
    <svg
      {...rest}
      className={clsx(styles.svg, styles[size], { [styles.stroked]: stroked }, { [styles.filled]: filled }, className)}
      width="1em"
      height="1em"
      viewBox={viewBox}
    >
      {children}
    </svg>
  );
};

export default SvgIcon;
