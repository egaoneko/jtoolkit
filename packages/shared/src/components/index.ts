export * from './inputs';
export * from './displays';
export * from './utils';
