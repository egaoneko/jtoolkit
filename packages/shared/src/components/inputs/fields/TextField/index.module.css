.wrapper {
  @apply w-full;
}

.container {
  @apply relative flex items-center gap-2;
  @apply box-border overflow-hidden select-none;
  @apply border-b;

  &.doubleExtraSmall {
    @apply py-1.5 px-3 gap-1;
  }

  &.extraSmall {
    @apply py-2 px-4 gap-1.5;
  }

  &.small {
    @apply py-2.5 px-5 gap-2;
  }

  &.medium {
    @apply py-3 px-6 gap-2.5;
  }

  &.large {
    @apply py-3.5 px-7 gap-3;
  }

  &.extraLarge {
    @apply py-4 px-8 gap-3.5;
  }

  &.doubleExtraLarge {
    @apply py-5 px-10 gap-4;
  }

  &.rounded {
    @apply rounded-full;
  }

  &.disabled {
    @apply opacity-30;
  }
}

.input {
  @apply min-w-0 flex-1;
  @apply bg-transparent focus:outline-none;

  .doubleExtraSmall & {
    @apply text-xs;
  }

  .extraSmall & {
    @apply text-sm;
  }

  .small & {
    @apply text-lg;
  }

  .medium & {
    @apply text-2xl;
  }

  .large & {
    @apply text-4xl;
  }

  .extraLarge & {
    @apply text-6xl;
  }

  .doubleExtraLarge & {
    @apply text-8xl;
  }
}

.filled {
  @apply bg-gray-100 rounded-md shadow;

  &.primary {
    @apply text-white;
    @apply bg-bgPrimaryDark hover:bg-bgPrimaryDarker;
    @apply dark:bg-dark-bgPrimaryDark hover:dark:bg-dark-bgPrimaryDarker;
  }

  &.secondary {
    @apply text-white;
    @apply bg-bgSecondaryDark hover:bg-bgSecondaryDarker;
    @apply dark:bg-dark-bgSecondaryDark hover:dark:bg-dark-bgSecondaryDarker;
  }

  &.success {
    @apply text-white;
    @apply bg-bgSuccessDark hover:bg-bgSuccessDarker;
    @apply dark:bg-dark-bgSuccessDark hover:dark:bg-dark-bgSuccessDarker;
  }

  &.danger {
    @apply text-white;
    @apply bg-bgDangerDark hover:bg-bgDangerDarker;
    @apply dark:bg-dark-bgDangerDark hover:dark:bg-dark-bgDangerDarker;
  }

  &.warning {
    @apply text-white;
    @apply bg-bgWarningDark hover:bg-bgWarningDarker;
    @apply dark:bg-dark-bgWarningDark hover:dark:bg-dark-bgWarningDarker;
  }

  &.info {
    @apply text-white;
    @apply bg-bgInfoDark hover:bg-bgInfoDarker;
    @apply dark:bg-dark-bgInfoDark hover:dark:bg-dark-bgInfoDarker;
  }
}

.outlined {
  @apply border rounded-md shadow;

  &.primary {
    @apply text-textPrimaryDarker border-bgPrimaryDarker hover:bg-bgPrimaryLighter/[0.05];
    @apply dark:text-dark-textPrimaryDarker dark:border-dark-bgPrimaryDarker hover:dark:bg-dark-bgPrimaryLighter/[0.05];
  }

  &.secondary {
    @apply text-textSecondaryDarker border-bgSecondaryDarker hover:bg-bgSecondaryLighter/[0.05];
    @apply dark:text-dark-textSecondaryDarker dark:border-dark-bgSecondaryDarker hover:dark:bg-dark-bgSecondaryLighter/[0.05];
  }

  &.success {
    @apply text-textSuccessDarker border-bgSuccessDarker hover:bg-bgSuccessLighter/[0.05];
    @apply dark:text-dark-textSuccessDarker dark:border-dark-bgSuccessDarker hover:dark:bg-dark-bgSuccessLighter/[0.05];
  }

  &.danger {
    @apply text-textDangerDarker border-bgDangerDarker hover:bg-bgDangerLighter/[0.05];
    @apply dark:text-dark-textDangerDarker dark:border-dark-bgDangerDarker hover:dark:bg-dark-bgDangerLighter/[0.05];
  }

  &.warning {
    @apply text-textWarningDarker border-bgWarningDarker hover:bg-bgWarningLighter/[0.05];
    @apply dark:text-dark-textWarningDarker dark:border-dark-bgWarningDarker hover:dark:bg-dark-bgWarningLighter/[0.05];
  }

  &.info {
    @apply text-textInfoDarker border-bgInfoDarker hover:bg-bgInfoLighter/[0.05];
    @apply dark:text-dark-textInfoDarker dark:border-dark-bgInfoDarker hover:dark:bg-dark-bgInfoLighter/[0.05];
  }
}

.text {
  &.primary {
    @apply text-textPrimaryDarker hover:bg-bgPrimaryLight/[0.1];
    @apply dark:text-dark-textPrimaryDarker hover:dark:bg-dark-bgPrimaryLight/[0.1];
  }

  &.secondary {
    @apply text-textSecondaryDarker hover:bg-bgSecondaryLight/[0.1];
    @apply dark:text-dark-textSecondaryDarker hover:dark:bg-dark-bgSecondaryLight/[0.1];
  }

  &.success {
    @apply text-textSuccessDarker hover:bg-bgSuccessLight/[0.1];
    @apply dark:text-dark-textSuccessDarker hover:dark:bg-dark-bgSuccessLight/[0.1];
  }

  &.danger {
    @apply text-textDangerDarker hover:bg-bgDangerLight/[0.1];
    @apply dark:text-dark-textDangerDarker hover:dark:bg-dark-bgDangerLight/[0.1];
  }

  &.warning {
    @apply text-textWarningDarker hover:bg-bgWarningLight/[0.1];
    @apply dark:text-dark-textWarningDarker hover:dark:bg-dark-bgWarningLight/[0.1];
  }

  &.info {
    @apply text-textInfoDarker hover:bg-bgInfoLight/[0.1];
    @apply dark:text-dark-textInfoDarker hover:dark:bg-dark-bgInfoLight/[0.1];
  }
}
