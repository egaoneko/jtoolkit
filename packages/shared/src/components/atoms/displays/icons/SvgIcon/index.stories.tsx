import { Story } from '@storybook/react';

import { Size } from 'src/enums/size';
import styles from 'src/styles/storybook.css';

import SvgIcon, { SvgIconProps } from './index';

export default {
  title: 'atoms/displays/icons/SvgIcon',
  component: SvgIcon,
};

const StandardTemplate: Story<SvgIconProps> = args => (
  <div className={styles.row}>
    <div className={styles.primary}>
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className={styles.secondary}>
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className={styles.success}>
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className={styles.danger}>
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className={styles.warning}>
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className={styles.info}>
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
  </div>
);

export const Standard = StandardTemplate.bind({});

Standard.args = {
  size: Size.MEDIUM,
  stroked: false,
  filled: true,
};
