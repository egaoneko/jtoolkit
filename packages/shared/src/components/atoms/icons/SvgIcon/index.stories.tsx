import { Story } from '@storybook/react';

import { Size } from '../../../../enums/size';

import SvgIcon, { SvgIconProps } from './index';

export default {
  title: 'atoms/icons/SvgIcon',
  component: SvgIcon,
};

const StandardTemplate: Story<SvgIconProps> = args => (
  <div className="flex gap-2">
    <div className="text-textPrimary">
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className="text-textSecondary">
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className="text-textSuccess">
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className="text-textDanger">
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className="text-textWarning">
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
    <div className="text-textInfo">
      <SvgIcon {...args}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    </div>
  </div>
);

export const Standard = StandardTemplate.bind({});

Standard.args = {
  size: Size.MEDIUM,
  stroked: false,
  filled: true,
};
