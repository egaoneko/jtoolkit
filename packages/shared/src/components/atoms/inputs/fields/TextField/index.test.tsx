import renderer, { ReactTestInstance } from 'react-test-renderer';

import { Color, Size } from 'src/enums';
import SvgIcon from 'src/components/atoms/displays/icons/SvgIcon';
import TextField from 'src/components/atoms/inputs/fields/TextField/index';
import { FieldVariant } from 'src/enums/field';

describe('TextField Snapshots', () => {
  const icon = (
    <SvgIcon>
      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
    </SvgIcon>
  );

  test('renders TextField with default options', async () => {
    const tree = renderer.create(<TextField />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).type).toBe('div');
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[0] as ReactTestInstance)
        .type,
    ).toBe('input');
  });

  test('renders TextField with default', async () => {
    const tree = renderer.create(<TextField defaultValue="Test" />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[0] as ReactTestInstance)
        .props.defaultValue,
    ).toBe('Test');
  });

  test('renders TextField with variant', async () => {
    const tree = renderer.create(<TextField variant={FieldVariant.STANDARD} />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).props.className).toContain(
      FieldVariant.STANDARD,
    );
  });

  test('renders TextField with size', async () => {
    const tree = renderer.create(<TextField size={Size.SMALL} />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).props.className).toContain(
      Size.SMALL,
    );
  });

  test('renders TextField with color', async () => {
    const tree = renderer.create(<TextField color={Color.SECONDARY} />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).props.className).toContain(
      Color.SECONDARY,
    );
  });

  test('renders TextField with rounded', async () => {
    const tree = renderer.create(<TextField rounded />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).props.className).toContain(
      'rounded',
    );
  });

  test('renders TextField with startIcon', async () => {
    const tree = renderer.create(<TextField startIcon={icon} />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[0] as ReactTestInstance)
        .type,
    ).toBe('span');
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[1] as ReactTestInstance)
        .type,
    ).toBe('input');
  });

  test('renders TextField with endIcon', async () => {
    const tree = renderer.create(<TextField endIcon={icon} />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[0] as ReactTestInstance)
        .type,
    ).toBe('input');
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[1] as ReactTestInstance)
        .type,
    ).toBe('span');
  });

  test('renders TextField with icons', async () => {
    const tree = renderer.create(<TextField startIcon={icon} endIcon={icon} />);
    expect(tree.toJSON()).toMatchSnapshot();
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[0] as ReactTestInstance)
        .type,
    ).toBe('span');
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[1] as ReactTestInstance)
        .type,
    ).toBe('input');
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[2] as ReactTestInstance)
        .type,
    ).toBe('span');
  });
});
