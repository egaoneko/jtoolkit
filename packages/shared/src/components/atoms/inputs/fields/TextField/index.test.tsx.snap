// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Snapshots renders TextField with color 1`] = `
<div
  className="wrapper"
>
  <div
    className="container medium standard secondary"
  >
    <input
      className="input"
      disabled={false}
    />
  </div>
</div>
`;

exports[`TextField Snapshots renders TextField with default 1`] = `
<div
  className="wrapper"
>
  <div
    className="container medium standard"
  >
    <input
      className="input"
      defaultValue="Test"
      disabled={false}
    />
  </div>
</div>
`;

exports[`TextField Snapshots renders TextField with default options 1`] = `
<div
  className="wrapper"
>
  <div
    className="container medium standard"
  >
    <input
      className="input"
      disabled={false}
    />
  </div>
</div>
`;

exports[`TextField Snapshots renders TextField with endIcon 1`] = `
<div
  className="wrapper"
>
  <div
    className="container medium standard"
  >
    <input
      className="input"
      disabled={false}
    />
    <span>
      <svg
        className="svg medium filled"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TextField Snapshots renders TextField with icons 1`] = `
<div
  className="wrapper"
>
  <div
    className="container medium standard"
  >
    <span>
      <svg
        className="svg medium filled"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </span>
    <input
      className="input"
      disabled={false}
    />
    <span>
      <svg
        className="svg medium filled"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`TextField Snapshots renders TextField with rounded 1`] = `
<div
  className="wrapper"
>
  <div
    className="container medium standard rounded"
  >
    <input
      className="input"
      disabled={false}
    />
  </div>
</div>
`;

exports[`TextField Snapshots renders TextField with size 1`] = `
<div
  className="wrapper"
>
  <div
    className="container small standard"
  >
    <input
      className="input"
      disabled={false}
    />
  </div>
</div>
`;

exports[`TextField Snapshots renders TextField with startIcon 1`] = `
<div
  className="wrapper"
>
  <div
    className="container medium standard"
  >
    <span>
      <svg
        className="svg medium filled"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </span>
    <input
      className="input"
      disabled={false}
    />
  </div>
</div>
`;

exports[`TextField Snapshots renders TextField with variant 1`] = `
<div
  className="wrapper"
>
  <div
    className="container medium standard"
  >
    <input
      className="input"
      disabled={false}
    />
  </div>
</div>
`;
